experiments/AtmosLES/riemann.jl

```diff
@@ -0,0 +1,345 @@
+# [here](https://docs.julialang.org/en/v1/manual/modules/#).
+
+# We need to use the `ClimateMachine` module! This imports all functions
+# specific to atmospheric and ocean flow modeling.
+
+using ClimateMachine
+ClimateMachine.init(parse_clargs = true)
+using ClimateMachine.Atmos
+using ClimateMachine.Orientations
+using ClimateMachine.ConfigTypes
+using ClimateMachine.Diagnostics
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.ODESolvers
+using ClimateMachine.TemperatureProfiles
+using ClimateMachine.Thermodynamics
+using ClimateMachine.TurbulenceClosures
+using ClimateMachine.VariableTemplates
+
+# In ClimateMachine we use `StaticArrays` for our variable arrays.
+# We also use the `Test` package to help with unit tests and continuous
+# integration systems to design sensible tests for our experiment to ensure new
+# / modified blocks of code don't damage the fidelity of the physics. The test
+# defined within this experiment is not a unit test for a specific
+# subcomponent, but ensures time-integration of the defined problem conditions
+# within a reasonable tolerance. Immediately useful macros and functions from
+# this include `@test` and `@testset` which will allow us to define the testing
+# parameter sets.
+using StaticArrays
+using Test
+using CLIMAParameters
+using CLIMAParameters.Atmos.SubgridScale: C_smag
+using CLIMAParameters.Planet: R_d, cp_d, cv_d, MSLP, grav
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet();
+
+# ## [Initial Conditions](@id init)
+# This example demonstrates the use of functions defined
+# in the [`Thermodynamics`](@ref Thermodynamics-docs) package to
+# generate the appropriate initial state for our problem.
+
+## !!! note
+##     The following variables are assigned in the initial condition
+##     - `state.ρ` = Scalar quantity for initial density profile
+##     - `state.ρu`= 3-component vector for initial momentum profile
+##     - `state.ρe`= Scalar quantity for initial total-energy profile
+##       humidity
+##     - `state.tracers.ρχ` = Vector of four tracers (here, for demonstration
+##       only; we can interpret these as dye injections for visualization
+##       purposes)
+function init_Riemann!(problem, bl, state, aux, (x, y, z), t)
+    ## Problem float-type
+    FT = eltype(state)
+    param = bl.param_set
+    ## Assign State Variables
+    ##Test2
+    #=if (x<0)
+      state.ρ = FT(1)
+      state.ρu = SVector{3,FT}(-2, 0, 0)
+      state.ρe = FT(3)
+      state.moisture.ρq_tot = FT(0)
+    else
+      state.ρ = FT(1)
+      state.ρu = SVector{3,FT}(2, 0, 0)
+      state.ρe = FT(3)
+      state.moisture.ρq_tot = FT(0)
+    end=#
+    ## Test 1
+    if (x < 0)
+        state.ρ = FT(1)
+        state.ρu = SVector{3, FT}(0, 0, 0)
+        p = FT(100000)
+        T = air_temperature_from_ideal_gas_law(param, p, FT(1))
+        state.ρe = internal_energy(param, T)
+        #state.moisture.ρq_tot = FT(0)
```

Comment:
We should use a thermo state constructor here. We can add a new one, using `ρ` and `p`, if that's what you need and if it makes sense. Same for below.